pkg/apis/pipeline/v1alpha1/pipeline_validation.go

```diff
@@ -32,6 +33,39 @@ func (p *Pipeline) Validate() *apis.FieldError {
 	return nil
 }
 
+func isOutput(task PipelineTask, resource string) bool {
+	for _, output := range task.Resources.Outputs {
+		if output.Resource == resource {
+			return true
+		}
+	}
+	return false
+}
+
+func validateDeclaredResources(ps *PipelineSpec) error {
+	needed := []string{}
```

Comment:
`s/needed/required/g` ?

Reply:
kk

Reply:
now these lines are exactly the same length, very satisfying :sunglasses: 

```
	required := []string{"elsa", "anna", "olaf", "kristoff"}
	provided := []string{"elsa", "anna", "olaf", "kristoff"}
```